saviSimulation/savi/asl/uav.asl

```diff
@@ -0,0 +1,110 @@
+/*
+ * UAV Agent Behaviour
+ * @author	Michael Vezina
+ * @date	26 March 2019
+ */
+
+// Set initial beliefs and test goals
+pi(3.14159265359).			// Set the constant for PI
+proximityThreshold(30.0).	// When the agent is closer than this threashold, no need to get closer
+
+/* Rules */
+// Define the turn angle
+turnAngle(A) :-
+	((A = PI/16) & pi(PI)).
+
+destLeft :-
+	(relativeDestination(AZ,EL,RANGE) &
+	turnAngle(ANGLE) &
+	pi(PI) &
+	((PI < AZ) &
+	(AZ < ((2 * PI) - ANGLE)))).
+
+destRight :-
+	(relativeDestination(AZ,EL,RANGE) &
+	turnAngle(ANGLE) &
+    	pi(PI) &
+    	(ANGLE < AZ) &
+    	(AZ <= PI)).
+
+destAhead(R) :-
+    (not destLeft & not destRight & relativeDestination(AZ, EL, R)).
+
+// Initial goals
+// TODO: Patrol general area vs just going to one destination
+!patrol.    // Patrol the map
+
+// Remove any beliefs broadcast by other agents so they don't litter the belief base!
++notifyThreat(_,_,_,_,_,_)[source(_)]
+    : true
+    <- -notifyThreat(_,_,_,_,_,_)[source(_)].
+
++!broadcastVisibleThreats
+    :   threat(AZ,EL,RANGE,RADIUS,TIME,TYPE) &
+        position(X_REF, Y_REF, Z_REF, _) &
+        velocity(BEARING, _,_,_)
+    <-  savi.UxVInternalActions.GetAbsolutePosition(X_TARGET,Y_TARGET,Z_TARGET,X_REF,Y_REF,Z_REF,BEARING,AZ,EL,RANGE);
+       	.broadcast(tell, notifyThreat(X_TARGET,Y_TARGET,Z_TARGET,RADIUS,TIME,TYPE)).
+
++!broadcastVisibleThreats.
+
+// Hard coded random generator between 100 and 800 (X value of destination)
+nextRand(X) :-
+    .random(R) & X = (R*700)+100.
```

Comment:
I get it now, this is random movement, similar to the threats. I'm curious if we can try and command the UAV to a destination by sending it the destination(...) message? Might make for a nice feature in your demo.

Reply:
Also, instead of the arbitrary boundaries, it may be a good idea for the agent's to have knowledge of the map size. Not sure how I would do that though, perhaps as a perception containing the locations of the map corners? Food for thought for later (not this PR).